src/Paymaster.sol

```diff
@@ -16,20 +16,20 @@ contract Paymaster{
         balances[token][msg.sender] += amount;
     }
 
-    function sponsorGas(IERC20 token, address tokenAddress, address from, address target) public {
+    function sponsorGas(IERC20 token, address tokenAddress, address from, address target, uint choice) public {
```

Comment:
`choice` looks like a boolean. also, you should give a more precise name, like `bool gasInEth`